File: python/stubs/stub_openai_client.py
from openai import OpenAI, Stream
from openai.types import ChatModel
from openai.types.chat import ChatCompletionMessageParam, ChatCompletionChunk
from openai.types.chat.chat_completion_chunk import Choice, ChoiceDelta
from typing import Generator
import time
from datetime import datetime


chunks = ["Hello ", "from ", "OpenAI!"]

def openai_chat_completion(model: ChatModel, messages: list[ChatCompletionMessageParam], temperature: float, max_tokens: float, stream: bool) -> Generator[ChatCompletionChunk, None, None]:
    for index, chunk in enumerate(chunks):
        yield ChatCompletionChunk(
            id=str(index),
            choices=[
                Choice(
                    index=0,
                    delta=ChoiceDelta(
                        role="assistant",
                        content=chunk
                    )
                )
            ],
            created=datetime.now().timestamp().is_integer(),
            model=model,
            object="chat.completion.chunk",
        )
        time.sleep(1)
    

File: typescript/src/stubs/stub-openai-client.ts
import { OpenAI } from "openai";
import { Stream } from "openai/streaming";

const chunks = ["Hello ", "from ", "OpenAI!"];

async function* stubGenerator(
  model: string
): AsyncGenerator<OpenAI.Chat.ChatCompletionChunk> {
  for (const [index, output] of chunks.entries()) {
    yield {
      id: index.toString(),
      choices: [
        {
          index: 0,
          delta: {
            role: "assistant",
            content: output,
          },
          finish_reason: null,
        },
      ],
      model: model,
      object: "chat.completion.chunk",
      created: Date.now(),
    };
  }
}

export function openAiChatCompletion(
  input: OpenAI.Chat.ChatCompletionCreateParamsStreaming
): {
  confidence: number;
  stream: Stream<OpenAI.Chat.ChatCompletionChunk>;
} {
  return {
    confidence: (Math.floor(Math.random() * 10) + 1) / 10,
    stream: new Stream(() => stubGenerator(input.model), new AbortController()),
  };
}
